### PR TITLE
Added check for name uniqueness

### PR DESCRIPTION
**Pull request by aklochkova**
Fixes https://github.com/OHDSI/Atlas/issues/1338

### CODE REVIEW

File: js/services/CohortDefinition.js [hunk 1]
```diff
@@ -65,7 +65,14 @@ define(function (require, exports) {
 				authApi.handleAccessDenied(error);
 			}
 		});
-		return loadPromise;	
+		return loadPromise;
+	}
+
+	function exists(name, id) {
+		return $.ajax({
```
Comment (wivern) [inline js/services/CohortDefinition.js]: use http service

File: js/services/IRAnalysis.js [hunk 2]
```diff
@@ -148,7 +148,17 @@ define(function (require, exports) {
 				authApi.handleAccessDenied(response);
 				return response;
 			});
-	}	
+	}
+
+	function exists(name, id) {
+		return httpService
+			.doGet(`${config.webAPIRoot}ir` + '/exists?name=' + name + '&id=' + id)
```
Comment (wivern) [inline js/services/IRAnalysis.js]: mix of string interpolation and concatenation

**Comment by pavgra** [comment]
@wivern / @aklochkova , let's be consistent and stick to async / await instead of Promises

File: js/pages/characterizations/components/characterizations/characterization-view-edit.js [hunk 3]
```diff
@@ -48,9 +48,13 @@ define([
 
             this.designDirtyFlag = sharedState.CohortCharacterization.dirtyFlag;
             this.loading = ko.observable(false);
-            this.isNameCorrect = ko.computed(() => {
+            this.defaultName = 'New Characterization';
+            this.isNameFilled = ko.computed(() => {
                 return this.design() && this.design().name();
             });
+            this.isNameCorrect = ko.computed(() => {
```
[the remark this answers is not in the document]
Comment (aklochkova) [reply js/pages/characterizations/components/characterizations/characterization-view-edit.js]: further in isSavePermittedResolver()

File: js/pages/prediction/prediction-manager.js [hunk 4]
```diff
@@ -199,17 +205,30 @@ define([
...
-				this.isSaving(false);
-				this.loading(false);
-			});
+
+			// Next check to see that a prediction analysis with this name does not already exist
+			// in the database. Also pass the id so we can make sure that the current prediction analysis is excluded in this check.
+			try{
+				const results = await PredictionService.exists(this.patientLevelPredictionAnalysis().name(), this.patientLevelPredictionAnalysis().id() == undefined ? 0 : this.patientLevelPredictionAnalysis().id());
+				if (results > 0) {
+					this.isSaving(false);
```
Comment (wivern) [inline js/pages/prediction/prediction-manager.js]: use finally block

**Comment by wivern** [comment]
You're using 3 different styles in services calls:
1. `doGet(config.webAPIRoot + predictionEndpoint + 'exists?name=' + name + '&id=' + id)`
1. `doGet(config.webAPIRoot + 'conceptset/exists', { name, id })`
1. `doGet(`${config.webAPIRoot}ir/exists?name=${name}&id=${id}`)`
Please, choose one of them and make all service calls in a single style.

File: js/pages/cohort-definitions/cohort-definition-manager.js [hunk 5]
```diff
@@ -657,33 +663,45 @@ define(['jquery', 'knockout', 'text!./cohort-definition-manager.html',
 					});
 			}
 
-			save () {
-			this.isSaving(true);
-			clearTimeout(this.pollTimeout);
-			this.model.clearConceptSet();
+			async save () {
+				this.isSaving(true);
+				clearTimeout(this.pollTimeout);
```
[the remark this answers is not in the document]
Comment (aklochkova) [reply js/pages/cohort-definitions/cohort-definition-manager.js]: it isn't removed, it is moved in 'else' block

File: js/components/conceptsetmodal/conceptSetSaveModal.js [hunk 6]
```diff
@@ -19,7 +19,7 @@ define(['knockout', 'appConfig', 'services/AuthAPI', 'services/ConceptSet', 'tex
         return self.isNameVerified() && !self.isNameUnique();
       });
       self.checkName = ko.computed(function () {
-        conceptSetService.exists(self.conceptSetName())
+        conceptSetService.exists(self.conceptSetName() === undefined ? '' : self.conceptSetName(), 0)
```
Comment (aklochkova) [inline js/components/conceptsetmodal/conceptSetSaveModal.js]: i don't get how it worked and what parameters should be here, so i use defaults

**Pull request merged**